Tasks/Grunt/grunttask.ts

```diff
@@ -41,11 +47,85 @@ gt.pathArg(gruntFile);
 
 gt.argString(tl.getInput('arguments', false));
 
-gt.exec()
-.then(function(code) {
-	tl.setResult(tl.TaskResult.Succeeded, tl.loc('GruntReturnCode', code));
-})
-.fail(function(err) {
+var enableCoverage = tl.getBoolInput('enableCodeCoverage');
+
+if (enableCoverage) {
+	var testSrc = tl.getPathInput('testFiles', true, false);
+
+	if (os.type().match(/^Win/)) {
+		var istanbul = tl.createToolRunner(tl.which('istanbul', true));
+	} else {
+		var istanbul = tl.createToolRunner(tl.which('./node_modules/istanbul/lib/cli.js', true));
+	}
+
+	istanbul.argString('cover --report cobertura');
+	istanbul.arg('./node_modules/mocha/bin/_mocha');
+	istanbul.arg(testSrc);
+	istanbul.argString('-- --ui bdd -t 5000');
+
+	var buildFolder = tl.getVariable('System.DefaultWorkingDirectory');
+
+	var summaryFile = path.join(buildFolder, 'coverage/cobertura-coverage.xml');
+}
+
+var publishJUnitResults = tl.getBoolInput('publishJUnitResults');
+var testResultsFiles = tl.getInput('testResultsFiles', publishJUnitResults);
+
+npm.exec().then(function () {
+	gt.exec()
+		.then(function (code) {
+			if (enableCoverage) {
+				istanbul.exec().then(function (code) {
+					publishTestResults(publishJUnitResults, testResultsFiles);
```

Comment:
can this move out of "pass/fail" block as we always call publishTestResults irrespective of "code coverage" functionality